app/src/main/java/com/github/geohunt/app/model/database/firebase/FirebaseDatabase.kt

```diff
@@ -133,13 +156,72 @@ class FirebaseDatabase(activity: Activity) : Database {
         return imageRefById.get(FirebaseBitmapRef.getImageIdFromChallengeId(cid))
     }
 
+    internal fun getProfilePicture(uid: String): FirebaseBitmapRef {
+        return imageRefById.get(FirebaseBitmapRef.getImageIdFromUserId(uid))
+    }
+
     internal fun getClaimRefById(id: String) : LazyRef<Claim> {
         TODO()
     }
 
     override fun getNearbyChallenge(location: Location): Task<List<Challenge>> {
-        TODO()
+        val listOfQuadrant = listOf("3cc359ec")
+
+        val xs = listOfQuadrant.map { cid ->
+            dbChallengeRef.child(cid).get()
+                .thenMap {
+                    val challenge = it.buildChallenge(this, cid)
+                    challengeRefById.register(cid, FirebaseChallengeRef(cid, this, challenge))
+                    challenge
+                }
+        }
+
+        return Tasks.whenAllSuccess<Challenge>(xs)
+            .thenMap { it.toImmutableList() }
+    }
+
+    override fun getLikesOf(uid: String): LazyRef<List<Challenge>> {
```

Comment:
Again, continuing from the previous comment, from your Implementation , one can retrieve the challenges a particular user likes. But not the list of users that like a given challenge, which I think is the goal of this task

Reply:
The same as above.

Reply:
As above, this has been changed.